src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1199,7 +1219,15 @@ else if (samplingLevel < indexSummary.getSamplingLevel())
             }
 
             // Always save the resampled index
-            saveSummary(newSummary);
+            tidy.global.writeLock.lock();
+            try
+            {
+                saveSummary(newSummary);
```

Comment:
@jasonstack It looks like this is the only summary saving path that could affect streaming, but wanted to make sure...

Reply:
yeah, this is the only one. I think the other index-summary modifications are used before being part of the node.